src/badfish/badfish.py

```diff
@@ -1958,7 +1958,7 @@ async def get_screenshot(self, gif=False):
         elif status_code == 404:
             raise BadfishException("The system does not support screenshots.")
         else:
-            if not gif:
+            if not gif or (gif and status_code == 400 and first):
```

Comment:
why do we need this only on the first one?
if it fails on the first one, execution is interrupted and BF exception is raised terminating the whole execution.
if we don't need to discriminate only for the first one we can just replace this line with:
`if not gif or status_code == 400:`

Reply:
We only need the first one, because when I tested it, sometimes when asking for a lot of screenshots in a row, a message with code 400 can come up, but the server was still on. 
For this reason, it can return None, and the function that makes the gif just ignores this one and continues with asking for screenshots, which it will get. 
So if we get it on the first one, either when creating a single screenshot or gif, we can assume that the server is off.

Reply:
I see your point. Fine to leave it there then just a small amendment:
`if not gif or (status_code == 400 and first):`

Reply:
Should also be fixed. 👍